src/query/src/dist_plan/merge_scan.rs

```diff
@@ -156,20 +156,19 @@ impl MergeScanExec {
         query_ctx: QueryContextRef,
         target_partition: usize,
     ) -> Result<Self> {
-        let arrow_schema_without_metadata = Self::arrow_schema_without_metadata(arrow_schema);
+        let arrow_schema = Arc::new(arrow_schema.clone());
         let properties = PlanProperties::new(
-            EquivalenceProperties::new(arrow_schema_without_metadata.clone()),
+            EquivalenceProperties::new(arrow_schema.clone()),
             Partitioning::UnknownPartitioning(target_partition),
             ExecutionMode::Bounded,
         );
-        let schema_without_metadata =
-            Self::arrow_schema_to_schema(arrow_schema_without_metadata.clone())?;
+        let schema = Self::arrow_schema_to_schema(arrow_schema.clone())?;
         Ok(Self {
             table,
             regions,
             plan,
-            schema: schema_without_metadata,
-            arrow_schema: arrow_schema_without_metadata,
+            schema,
+            arrow_schema,
             region_query_handler,
             metric: ExecutionPlanMetricsSet::new(),
             sub_stage_metrics: Arc::default(),
```

Comment:
Let's introduce a `remove_column_metadata` method for `Schema`

```rust
    pub fn remove_column_metadata(mut self) -> Schema {
        self.column_schemas = self
            .column_schemas
            .into_iter()
            .map(|column_schema| column_schema.with_metadata(Default::default()))
            .collect();

        self.arrow_schema = Arc::new(ArrowSchema::new(
            self.arrow_schema
                .fields()
                .iter()
                .map(|field| {
                    let field = field.as_ref().clone();
                    let field_without_metadata = field.with_metadata(Default::default());
                    Arc::new(field_without_metadata)
                })
                .collect::<Vec<_>>(),
        ));

        self
    }
```

then you can use it like: 

```rust
let schema = Self::arrow_schema_to_schema(arrow_schema.clone())?.remove_column_schema();
let arrow_schema = schema.arrow_schema().clone();

...
```